docs/metrics.md

```diff
@@ -46,4 +58,213 @@ metrics, to prevent expanding the required set of dependencies. Please see [armo
\ No newline at end of file
+For targeted attacks, each metric will be reported twice for adversarial data: once relative to the ground truth labels and once relative to the target labels.  For untargeted attacks, each metric is only reported relative to the ground truth labels.  Performance relative to ground truth measures the effectiveness of the defense, indicating the ability of the model to make correct predictions despite the perturbed input.  Performance relative to target labels measures the effectiveness of the attack, indicating the ability of the attacker to force the model to make predictions that are not only incorrect, but that align with the attackers chosen output.
+
+## Instrumentation
+
+The `armory.instrument` module implements functionality to flexibly capture values for measurement.
+
+The primary mechanisms are largely based off of the logging paradigm of loggers and handlers, though with significant differences on the handling side.
+
+Probe - object to capture data and publish them for measurement.
+Meter - object to measure a single metric with given captured data and output records
+Writer - object to take meter output records and send them standard outputs (files, loggers, results dictionaries, etc.)
+Hub - object to route captured probe data to meter inputs and route meter outputs to writers
+There is typically only a single hub, where there can be numerous of the other types of objects.
+
+### Probes
+
+To get a new Probe (connected to the default Hub):
+```
+# Module imports section
+from armory.instrument import get_probe
+probe = get_probe(name)
+```
+The arg `name` can be any `str` that is a valid python identifier, or can be blank, which defaults to the empty string `""`.
+Similarly to `logging.getLogger`, this provides a namespace to place variables, and inputs like `__name__` can also be used.
+Calls to `get_probe` using the same name will return the same Probe object.
+The recommended approach is to set a probe at the top of the file of interest and use it for all captures in that file.
+
+To capture values in-line, use `update`:
+```
+# ...
+# In the code
+probe.update(name=value)
+```
+
+This will publish the given value(s) to the given name(s) (also called probe variables) in the probe namespace of the connected Hub.
+To be more concrete:
+```
+probe = get_probe("my.probe_name")
+probe.update(arbitrary_variable_name=15)
+```
+will push the value 15 to `"my.probe_name.arbitrary_variable_name"`.
+These names will be used when instantiating `Meter` objects.
+
+However, this will fall on the floor (`del`, effectively) unless a meter is connected to the Hub to record values.
```

Comment:
I think it would be most clear/effective if immediately after we say "this will fall on the floor unless you do XYZ", we show an example of exactly how to do XYZ

Reading these docs, it seems very clear to me how to set up a probe but it's less obvious how to actually do anything with the value I'm storing. Perhaps even just rearranging the order of different pieces of the docs would make this more clear

Reply:
I added a Quick Start example to the previous section and clarified this line.